ACM-18041: Adjust `ConfigurationPolicy` `customMessage` doc (GRC) (#7521)
open-cluster-management/rhacm-docs: governance/config_policy_ctrl.adoc

If you have existing Kubernetes manifests that you want to put in a policy, the Policy Generator is a useful tool to accomplish this.

[#configuration-policy-yaml]

> == Configuration policy YAML structure

Reviewer: ```suggestion
== Configuring a policy YAML structure
```

Author: Similar to above. This one is our YAML reference for `ConfigurationPolicy`. Not sure how to state that with a gerund 🤔

> *NOTE:* You can find this information by issuing an `oc explain --api-version=policy.open-cluster-management.io/v1 ConfigurationPolicy.<field-path>` command, replacing `<field-path>` with the path to the field you're looking for.

Reviewer: I think this content would work better as an intro statement rather than a Note, what do you think? Also, I'm not sure how this content relates to the title, "Configuration policy YAML structure" 

This content is almost more like a procedure and action steps we want the user to complete. So, maybe we can make these steps and add some context about how finding a filed helps configure a policy YAML structure? 

```suggestion
To find a field, issue an `oc explain --api-version=policy.open-cluster-management.io/v1 ConfigurationPolicy.<field-path>` command and replace `<field-path>` with the path to the field need.

For an example, see the following YAML sample: 
```

Author: Sure, I'll make it an intro statement!

This is our API reference rather than a set of steps (though in the future we'd like to have sets of steps about composing policies, but that's for another time). Using `oc explain` will return similar information as that in this table.

> To find the description of a field on your managed cluster, issue an `oc explain --api-version=policy.open-cluster-management.io/v1 ConfigurationPolicy.<field-path>` command and replace `<field-path>` with the path to the field needed.

Reviewer: I'm not opposed to using the verb "issue" here, but we typically say "run a command" is there a  specific reason why we'd want to say "issue a command" here? 

 ```suggestion
To find the description of a field on your managed cluster, run an `oc explain --api-version=policy.open-cluster-management.io/v1 ConfigurationPolicy.<field-path>` command and replace `<field-path>` with the path to the field needed.
```

Author: Nah, just the wording I put in there--I'll update it. 😄

> To find the description of a field on your managed cluster, issue an `oc explain --api-version=policy.open-cluster-management.io/v1 ConfigurationPolicy.<field-path>` command and replace `<field-path>` with the path to the field needed.

Reviewer: ```suggestion
To find the description of a field on your managed cluster, issue an `oc explain --api-version=policy.open-cluster-management.io/v1 ConfigurationPolicy.<field-path>` command and replace `<field-path>` with the path to the field that you need. 
```

> To find the description of a field on your managed cluster, run the command `oc explain --api-version=policy.open-cluster-management.io/v1 ConfigurationPolicy.<field-path>`, and replace `<field-path>` with the path to the field that you need.

Reviewer 2: This is a step but I see that the title was just changed to indicate a reference instead of a procedure so I'm a bit torn on this sentence. Normally, I'd change it to something like this: 

```suggestion
. Find the description of a field on your managed cluster by running the following command:

+
[source,bash]
---- 
oc explain --api-version=policy.open-cluster-management.io/v1 ConfigurationPolicy.<field-path> <1>
----
<1> Replace `<field-path>` with the path to the field that you need.
```

However, this would completely imply that this is a procedure. You could also say:
```suggestion
You can find the description of a field on your managed cluster by running the `oc explain --api-version=policy.open-cluster-management.io/v1 ConfigurationPolicy.<field-path>` command. Replace `<field-path>` with the path to the field that you need.
```

    noncompliant: {}
  severity: low
  evaluationInterval:
    compliant: ""
    noncompliant: ""
  object-templates-raw: ""
  object-templates: <2>
   - complianceType: musthave
     metadataComplianceType:
     recordDiff: ""
     recreateOption: ""
     objectSelector:
       matchLabels: {}
       matchExpressions: []
     objectDefinition:
       apiVersion: v1
       kind: Pod
       metadata:
         name: pod
       spec:
         containers:
          - image: pod-image
            name: pod-name
            ports:
             - containerPort: 80
   - complianceType: mustonlyhave
     objectDefinition:
       apiVersion: v1
       kind: ConfigMap
       metadata:
         name: myconfig
         namespace: default
         data:
           testData: hello
----
<1> When the `remediationAction` for the configuration policy is set to `enforce`, the controller applies the specified configuration to the target managed cluster. However, configuration policies that specify an object without a name can only be set to `inform`, unless the `objectSelector` is also configured.

> <2> A Kubernetes object is defined in the `object-templates` array in the configuration policy, where fields of the configuration policy controller is compared with objects on the managed cluster. You can also use templated values within configuration policies or, for more advanced use cases, alternatively specify a string in `object-templates-raw` to dynamically generate the desired `object-templates`. For more information, see _Template processing_.

Reviewer: Minor tweaks to match more the RH SG

```suggestion
<2> A Kubernetes object is defined in the `object-templates` array in the configuration policy, where fields of the configuration policy controller is compared with objects on the managed cluster. You can also use templated values within configuration policies. For more advanced use cases,  specify a string in `object-templates-raw` to create `object-templates` that you want. For more information, see _Template processing_.
```

[#configuration-policy-yaml-table]

> == Configuration policy YAML table

Reviewer: ```suggestion
== Configuring a policy YAML table
```

Author: Similar to above. This one is our YAML reference for ConfigurationPolicy. Not sure how to state that with a gerund 🤔

| `spec.customMessage`
| Optional

> | Use this section to configure the compliance message sent by the configuration policy based on the current compliance. Each message configuration is a string that can contain Go templates. The context variables `.DefaultMessage` and `.Policy` are available for use in the templates. You can access the default message using the `.DefaultMessage` parameter. The `.Policy` context variable contains the current policy object, including its status. For example, you can access the state of each related object by specifying `.Policy.status.relatedObjects[*].object`. If you set an `evaluationInterval` other than `watch`, only the kind, name, and namespace of the related objects are available.

Reviewer 2: ```suggestion
| Configure the compliance message sent by the configuration policy based on the current compliance. Each message configuration is a string that can contain Go templates. The `.DefaultMessage` and `.Policy` context variables are available for use in the templates. You can access the default message by using the `.DefaultMessage` parameter. The `.Policy` context variable contains the current policy object, including its status. For example, you can access the state of each related object by specifying the `.Policy.status.relatedObjects[*].object` field. If you set a value for the `evaluationInterval` field other than `watch`, only the kind, name, and namespace of the related objects are available.
```

| `spec.customMessage.compliant`
| Optional

> | Use this field to configure custom messages for configuration policies that are compliant. Go templates and UTF-8 encoded characters, including emoji and foreign characters, are supported values.

Reviewer 2: ```suggestion
| Configure custom messages for configuration policies that are compliant. Go templates and UTF-8 encoded characters, including emoji and foreign characters, are supported values.
```

| `spec.customMessage.noncompliant`
| Optional

> | Use this field to configure custom messages for configuration policies that are non-compliant. Go templates and UTF-8 encoded characters, along with emoji and foreign characters, are supported values.

Reviewer 2: ```suggestion
| Configure custom messages for configuration policies that are non-compliant. Go templates and UTF-8 encoded characters, including emoji and foreign characters, are supported values.
```
